tests/unit/test_gbq.py

```diff
@@ -505,3 +520,54 @@ def test_read_gbq_calls_tqdm(mock_bigquery_client, mock_service_account_credenti
 
     _, to_dataframe_kwargs = mock_list_rows.to_dataframe.call_args
     assert to_dataframe_kwargs["progress_bar_type"] == "foobar"
+
+
+def test_read_gbq_with_full_table_id(
+    mock_bigquery_client, mock_service_account_credentials
+):
+    mock_service_account_credentials.project_id = "service_account_project_id"
+    df = gbq.read_gbq(
+        "my-project.my_dataset.read_gbq_table",
+        credentials=mock_service_account_credentials,
+        project_id="param-project",
+    )
+    assert df is not None
+
+    mock_bigquery_client.query.assert_not_called()
+    mock_bigquery_client.list_rows.assert_called_with(
+        bigquery.Table("my-project.my_dataset.read_gbq_table"), max_results=None,
```

Comment:
Table comparison by reference-only was not added until a later version of google-cloud-bigquery. Need to update this and the other tests for the `unit-3.7` session